docs/segmentedtickerplant.md

```diff
@@ -0,0 +1,340 @@
+
+```
+// For single connections e.g. $TORQHOME/appconfig/settings/rdb.q
+.servers.CONNECTIONS:enlist `segmentedtickerplant
+
+// For multiple connections e.g. $TORQHOME/appconfig/settings/wdb.q 
+.servers.CONNECTIONS:`segmentedtickerplant`sort`gateway`rdb`hdb
+```
+
+The STP sets these functions in subscriber processes (these vary from process to process and by STP batching mode):
+- upd[t;x;now]
+  Called for updates from the STP. Arguments are t (the table the data is for), x (the data to be inserted) and now (the current time .z.p)
+- endofperiod[currentpd;nextpd;data]
+  Called at the end of a period for periodic STP modes. Takes 3 arguments currentpd (the current period), nextpd (the next period) and data (a table containing some basic information on the stp process and the current time on the stp)
+- endofday[date;data]
+  Called at the end of day for all modes. Takes 2 arguments date (current date) and data (a table containing some basic information on the stp process and the current time on the stp)
+
+**Error Trapping**
+
+If the `.stplg.errmode` Boolean variable is set to true, an error log is opened on start up and the `.u.upd` function is wrapped in an error trap, so that if a bad message is received, it is not published but instead sent to the error log. The advantage of this is that bad updates are not sent through or replayed into the subscribers, which could cause issues, and they are easier to find and debug.
+
+This mode is really designed for development/testing purposes, it shouldn't be necessary for a stable production feed and will add a small overhead to each update.
+
+**Time Zone Behaviour**
+
+A key tickerplant function is to timestamp the incoming data before it gets published to the rest of the system. In order to maintain applications and data in different time zones, such as a UK-based application which processes US data, the STP works with the `.eodtime` namespace variables in much the same way as other TorQ processes. The key variable used is `.eodtime.dailyadj` but more information on setting up a TorQ process in a different time zone can be found [here](https://aquaqanalytics.github.io/TorQ/utilities/#eodtimeq).
+
+**Customisation and Flexibility**
+
+The STP has been designed with customisation in mind. To this end here are a couple of ways to tailor the process to suit a particular application. The first is utilising the fact that each table has its own UPD function, meaning that some additional processing, such as adding a sequence number or a time-zone offset, can be done in the STP itself rather than needing to be done in a separate process. This is done by altering the `.stplg.updtab` dictionary in the segmentedtickerplant settings config file:
```

Comment:
```suggestion
Each table has its own upd function, meaning that some additional processing, such as adding a sequence number or a time-zone offset, can be done in the STP itself rather than needing to be done in a separate process. This is done by altering the `.stplg.updtab` dictionary in the segmentedtickerplant settings config file. The default behaviour is for every update to automatically have the current timestamp applied to it.
```